### PR TITLE
Add has has a review field on instance model

### PR DESCRIPTION
**Pull request by lincmba**
Updates upon SubmissionReview creation
Tests

### CODE REVIEW

File: onadata/apps/logger/migrations/0054_instance_has_a_review.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+    dependencies = [
+        ('logger', '0053_submissionreview'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='instance',
+            name='has_a_review',
+            field=models.BooleanField(default=False, verbose_name='Has Had A Review'),
```
Comment (moshthepitt) [inline onadata/apps/logger/migrations/0054_instance_has_a_review.py]: why not `Has Had A Review`

File: onadata/apps/logger/models/submission_review.py [hunk 2]
```diff
@@ -16,6 +16,8 @@ def update_instance_json_on_save(sender, instance, **kwargs):
     Signal handler to update Instance Json with the submission review on save
     """
     submission_instance = instance.instance
+    if not submission_instance.has_a_review:
+        submission_instance.has_a_review = True
     submission_instance.save()
```
Comment (moshthepitt) [inline onadata/apps/logger/models/submission_review.py]: does .save() need to be within the if block?

File: onadata/apps/logger/models/instance.py [hunk 3]
```diff
@@ -335,10 +335,9 @@ def get_full_dict(self, load_existing=True):
             if isinstance(self.deleted_at, datetime):
                 doc[DELETEDAT] = self.deleted_at.strftime(MONGO_STRFTIME)
 
-            if self.get_review_status():
+            # pylint: disable=E1101
```
Comment (moshthepitt) [inline onadata/apps/logger/models/instance.py]: Please write the English name of the rule you are disabling (for readability) is there is space on the line

File: onadata/apps/logger/models/submission_review.py (hunk 2 above)
Comment (lincmba) [reply onadata/apps/logger/models/submission_review.py]: Not really. It still needs to be called even when `has_a_review = True`, to update with the latest review incase of multiple reviews.

**Comment by ukanga** [comment]
Nevermind I think I have traced it. 

:+1:

**Pull request merged**